From 47ecba971672d28828456ec18f25edc237e255b7 Mon Sep 17 00:00:00 2001
From: Joe Fiorini <joe@faithfulgeek.org>
Date: Sat, 20 Feb 2010 11:29:11 -0500
Subject: [PATCH] Require bundled gems after loading Application

Some bundled gems may depend on Application config
settings (such as Rails.root). These are not defined
until after the Application class is loaded, therefore
we shouldn't require the gems until the class is set up.
---
 .../rails/app/templates/config/application.rb      |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/railties/lib/generators/rails/app/templates/config/application.rb b/railties/lib/generators/rails/app/templates/config/application.rb
index 7c555c2..1d7afc7 100644
--- a/railties/lib/generators/rails/app/templates/config/application.rb
+++ b/railties/lib/generators/rails/app/templates/config/application.rb
@@ -11,9 +11,6 @@ require "active_resource/railtie"
 require "rails/test_unit/railtie"
 <% end -%>
 
-# Auto-require default libraries and those for the current Rails environment.
-Bundler.require :default, Rails.env
-
 module <%= app_const_base %>
   class Application < Rails::Application
     # Settings in config/environments/* take precedence over those specified here.
@@ -49,3 +46,7 @@ module <%= app_const_base %>
     config.filter_parameters << :password
   end
 end
+
+# Auto-require default libraries and those for the current Rails environment.
+Bundler.require :default, Rails.env
+
-- 
1.6.4.4

